File: Service-Lane.cpp
vector<int> serviceLane(int n, vector<vector<int>> cases, vector<int> width)
{
    vector<int> result;
    
    for (size_t i = 0; i < cases.size(); ++i)
    {    
        int min = INT_MAX;
        int start = cases[i][0];
        int end = cases[i][1];
        
        for (int j = start; j <= end; ++j)
        {
            if (width[j] < min)
            {
                min = width[j];
            }
        }
        
        result.emplace_back(min);
    }
    
    return result;
}

vector<int> serviceLane1(int n, vector<vector<int>> cases, vector<int> width)
{
    vector<int> result;
    
    for (size_t i = 0; i < cases.size(); ++i)
    {   
        int start = cases[i][0];
        int end = cases[i][1];
        
        result.emplace_back(
            *min_element(width.begin()+start, width.end()-width.size()+end+1));       
    }
    
    return result;
}

File: Service-Lane.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'serviceLane' function below.
     *
     * The function is expected to return an INTEGER_ARRAY.
     * The function accepts following parameters:
     *  1. INTEGER n
     *  2. 2D_INTEGER_ARRAY cases
     */

    public static List<int> serviceLane(int n, List<List<int>> cases, List<int> width)
    {
        List<int> result = new List<int>();

        for (int i = 0; i < cases.Count; i++)
        {
            int start = cases[i][0];
            int end = cases[i][1];

            result.Add(width.GetRange(start, end - start + 1).Min());
        }

        return result;
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string[] firstMultipleInput = Console.ReadLine().TrimEnd().Split(' ');

        int n = Convert.ToInt32(firstMultipleInput[0]);

        int t = Convert.ToInt32(firstMultipleInput[1]);

        List<int> width = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(widthTemp => Convert.ToInt32(widthTemp)).ToList();

        List<List<int>> cases = new List<List<int>>();

        for (int i = 0; i < t; i++)
        {
            cases.Add(Console.ReadLine().TrimEnd().Split(' ').ToList().Select(casesTemp => Convert.ToInt32(casesTemp)).ToList());
        }

        List<int> result = Result.serviceLane(n, cases, width);

        textWriter.WriteLine(String.Join("\n", result));

        textWriter.Flush();
        textWriter.Close();
    }
}
